tests/integration/kubernetes/k8s-guest-pull-image-authenticated.bats

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env bats
+# Copyright (c) 2024 IBM Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/lib.sh"
+load "${BATS_TEST_DIRNAME}/confidential_common.sh"
+load "${BATS_TEST_DIRNAME}/confidential_kbs.sh"
+
+export KBS="${KBS:-false}"
+
+setup() {
+    # Log checking not working on TDX
+    if [ "${KATA_HYPERVISOR}" = "qemu-tdx" ]; then
+	    skip "Test skipped on ${KATA_HYPERVISOR}, see https://github.com/kata-containers/kata-containers/issues/10011"
+    fi
+
+    if ! is_confidential_runtime_class; then
+        skip "Test not supported for ${KATA_HYPERVISOR}."
+    fi
+
+    [ "${SNAPSHOTTER:-}" = "nydus" ] || skip "None snapshotter was found but this test requires one"
+
+    setup_common
+    AUTHENTICATED_IMAGE="${AUTHENTICATED_IMAGE:-quay.io/kata-containers/confidential-containers-auth:test}"
+    AUTHENTICATED_IMAGE_USER=${AUTHENTICATED_IMAGE_USER:-}
+    AUTHENTICATED_IMAGE_PASSWORD=${AUTHENTICATED_IMAGE_PASSWORD:-}
```

Comment:
Is it fine to continue execution with empty user or password ?

Reply:
In setup_kbs_credentials we skip the test if they are left empty

Reply:
What about https://github.com/kata-containers/kata-containers/pull/9904/files#diff-86f83bd393eb3acbe6e65f026a20c5b81a8528d9801bc99c12514a1d44ea6e78R32-R33 in the same `setup()` function ?

```
kubectl create secret docker-registry cococred --docker-server=https://quay.io/kata-containers/confidential-containers-auth --docker-username="" --docker-password=""
error: either --from-file or the combination of --docker-username, --docker-password and --docker-server is required
```
Same outcome if any of username, password of server is `""`  actually.

What about checking username and password here and provide a better error to the user ?

Reply:
Agreed - fix pushed :) Thank you